pages/auth-redirect.tsx

```diff
@@ -0,0 +1,25 @@
+import { CircularProgress } from '@mui/material';
+import axios from 'axios';
+import { useRouter } from 'next/router';
+import React from 'react';
+
+const AuthRedirect: React.FC = () => {
+  const router = useRouter();
+
+  const useAuthCode = async (code: string) => {
+    await axios.post('/api/auth/login', { code });
+    router.push('/');
+  };
+
+  React.useEffect(() => {
+    if (router.query.code) {
+      useAuthCode(String(router.query.code));
+    }
+  }, [router.query.code]);
+
+  return (
+    <div><CircularProgress /></div>
+  );
+};
+
+export default AuthRedirect;
```

Comment:
This is the page that users are redirected to after authenticating with Discord - Discord gives us a code, which we then pass to the server to fetch their Discord info.  Once successful, it'll redirect back to the home page.